z3rro/src/prover.rs

```diff
@@ -21,6 +50,65 @@ pub enum ProveResult {
     Unknown(ReasonUnknown),
 }
 
+/// Execute swine on the file located at file_path
+fn execute_swine(file_path: &Path) -> Result<(SatResult, String), ProverCommandError> {
+    let output = Command::new("swine").arg(file_path).output();
+
+    match output {
+        Ok(output) => {
+            let stdout = String::from_utf8_lossy(&output.stdout);
+            let mut lines_buffer: VecDeque<&str>= stdout.lines().collect();
+            let first_line = lines_buffer.pop_front().ok_or(ProverCommandError::ParseError)?;
+            if first_line.trim().to_lowercase() == "unsat" {
+                Ok((SatResult::Unsat, "".to_string()))
+            } else if first_line.trim().to_lowercase() == "sat" {
+                let _last_line = lines_buffer.pop_back().ok_or(ProverCommandError::ParseError)?;
+                if _last_line.contains("SHA") {
+                    let cex = lines_buffer.iter().join("");
+                    Ok((SatResult::Sat, cex))
+                } else {
+                    Err(ProverCommandError::ParseError)
+                }
+            } else {
```

Comment:
We should only return `Unknown` if the SAT solver result is actually unknown. If there's a different result than we expected, we must throw an error.